rtgui/editcallbacks.h

```diff
@@ -0,0 +1,256 @@
+/*
+ *  This file is part of RawTherapee.
+ *
+ *  Copyright (c) 2019 Jean-Christophe FRISCH <natureh.510@gmail.com>
+ *
+ *  RawTherapee is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  RawTherapee is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with RawTherapee.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "editid.h"
+#include "cursormanager.h"
+#include "../rtengine/coord.h"
+
+class Geometry;
+class EditDataProvider;
+
+/** @file
+ * See editwidgets.h for documentation
+ */
+
+/// @brief Method for client tools needing Edit information
+class EditSubscriber
+{
+
+public:
+
+private:
+    EditUniqueID ID; /// this will be used in improcfun to locate the data that has to be stored in the buffer; it must be unique in RT
+    EditType editingType;
+    BufferType bufferType;
+    EditDataProvider *provider;

```

Comment:
Who owns `provider`? If it's `EditSubscriber` and `getEditProvider()` has to return a mutable instance then better use a `std::shared_ptr<>`.

If the provider isn't owned by `EditSubscriber` but already exists when the subscriber is created, why not provide it in the c'tor and store a `const &`?

Reply:
@Floessie `provider` is owned by the `EditDataProvider`, and setting this parameter through `setEditProvider` is connecting the two so they communicate, i.e. when you press an Edit button somewhere.